boards/common/openocd-nrf5.board.cmake

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Infer nrf51 vs nrf52 etc from the BOARD name. This enforces a board
+# naming convention: "nrf5x" must appear somewhere in the board name
+# for this to work.
+#
+# Boards which don't meet this convention can set this variable before
+# including this script.
+if (NOT DEFINED OPENOCD_NRF5_SUBFAMILY)
+  string(REGEX MATCH nrf5. OPENOCD_NRF5_SUBFAMILY "${BOARD}")
+endif()
+if("${OPENOCD_NRF5_SUBFAMILY}" STREQUAL "")
+  message(FATAL_ERROR
+    "Can't match nrf5 subfamily from BOARD name. "
+    "To fix, set CMake variable OPENOCD_NRF5_SUBFAMILY.")
+endif()
+
+# We can do the right thing for supported subfamilies using a generic
+# script, at least for openocd 0.10.0 and the openocd shipped with
+# Zephyr SDK 0.10.3.
+set(pre_init_cmds
+  "set WORKAREASIZE 0x4000"	# 16 kB RAM used for flashing
+  "source [find interface/jlink.cfg]"
```

Comment:
@mbolivar Would it be possible to make interface configurable and support cmsis-dap?
For example my changes to get it working on reel board:
```diff
diff --git a/boards/arm/reel_board/board.cmake b/boards/arm/reel_board/board.cmake
index 0dd66bc12a..f59867e3fa 100644
--- a/boards/arm/reel_board/board.cmake
+++ b/boards/arm/reel_board/board.cmake
@@ -1,6 +1,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
+set(OPENOCD_NRF5_SUBFAMILY "nrf52")
 board_runner_args(jlink "--device=nrf52" "--speed=4000")
 include(${ZEPHYR_BASE}/boards/common/pyocd.board.cmake)
 include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/openocd-nrf5.board.cmake)
diff --git a/boards/common/openocd-nrf5.board.cmake b/boards/common/openocd-nrf5.board.cmake
index a378b00398..ba7deea98d 100644
--- a/boards/common/openocd-nrf5.board.cmake
+++ b/boards/common/openocd-nrf5.board.cmake
@@ -20,7 +20,7 @@ endif()
 # Zephyr SDK 0.10.3.
 set(pre_init_cmds
   "set WORKAREASIZE 0x4000"    # 16 kB RAM used for flashing
-  "source [find interface/jlink.cfg]"
+  "source [find interface/cmsis-dap.cfg]"
   "transport select swd"
   "source [find target/${OPENOCD_NRF5_SUBFAMILY}.cfg]"
   "$_TARGETNAME configure -rtos auto"
```

Reply:
@jfischer-phytec-iot sure, just add a variable for the interface, like OPENOCD_NRF5_INTERFACE, and source it instead of interface/jlink.cfg :)

Feel free to propose a patch and I will +1